_source/_posts/2021-10-04-spring-boot-spa.md

```diff
@@ -0,0 +1,646 @@
+---
+layout: blog_post
+title: "Learn How to Build a Single-Page App with Vue and Spring Boot"
+author: andrew-hughes
+by: contractor
+communities: [java,javascript]
+description: ""
+tags: [vue, java, spring-boot, spring, oauth2, oidc, spa, javascript]
+tweets:
+- ""
+- ""
+- ""
+image: blog/spring-boot-spa/spring-boot-spa-social.jpg
+type: conversion
+---
+
+In this tutorial, you are going to create a single-page application (SPA) that uses a Spring Boot resource server and a Vue front-end client. You'll see how to configure Spring Boot to use JSON Web Tokens (JWT) for authentication and authorization, with Okta as an OAuth 2. 0 and OpenID Connect (OIDC) provider. You'll also see how to bootstrap a Vue client app with the Vue CLI and how to secure it using the Okta Sign-In Widget.
+
+Okta is a computer security services company that provides a lot of great resources for securing web applications. The widget is a great way to secure front-end applications because it allows you to easily add a secure login form that is configurable for single sign-on and social sign-on with external providers such as Google, Facebook, and LinkedIn. It provides an implementation of the authorization code OAuth 2.0 flow using PKCE (Proof Key for Code Exchange).
```

Comment:
```suggestion
Okta is a computer security services company that provides a lot of great resources for securing web applications. The Okta Sign-In Wdiget is a great way to secure front-end applications because it allows you to easily add a secure login form that is configurable for single sign-on and social sign-on with external providers such as Google, Facebook, and LinkedIn. It provides an implementation of the authorization code OAuth 2.0 flow using PKCE (Proof Key for Code Exchange).
```